### PR TITLE
{Compute} Fix tests

### PR DESCRIPTION
**Pull request by qwordy**
**Description**

Fix

azure/cli/command_modules/vm/tests/latest/test_vm_commands.py::VMManagedDiskScenarioTest::test_vm_disk_max_shares_etc
azure/cli/command_modules/vm/tests/latest/test_vm_commands.py::DiskAccessTest::test_disk_access
azure/cli/command_modules/vm/tests/latest/test_vm_commands.py::VMGalleryImage::test_gallery_specialized

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 1]
```diff
@@ -3230,7 +3230,7 @@ def create_gallery_image(cmd, resource_group_name, gallery_name, gallery_image_n
                          os_type=os_type, os_state=os_state, end_of_life_date=end_of_life_date,
                          recommended=recommendation, disallowed=Disallowed(disk_types=disallowed_disk_types),
                          purchase_plan=purchase_plan, location=location, eula=eula, tags=(tags or {}),
-                         hyper_vgeneration=hyper_v_generation, features=feature_list)
+                         hyper_v_generation=hyper_v_generation, features=feature_list)
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:3233]: is this due to track2?

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 2]
```diff
@@ -770,9 +770,9 @@ def test_vm_disk_max_shares_etc(self, resource_group):
         self.cmd('sig create -g {rg} --gallery-name {g1}')
         self.cmd('sig image-definition create -g {rg} --gallery-name {g1} --gallery-image-definition image --os-type linux -p publisher1 -f offer1 -s sku1 --features "IsSecureBootSupported=false IsMeasuredBootSupported=false" --hyper-v-generation V2', checks=[
             self.check('features[0].name', 'IsSecureBootSupported'),
-            self.check('features[0].value', 'false'),
+            self.check('features[0].value', 'false', False),
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:773]: why the change

File: src/azure-cli/azure/cli/command_modules/vm/custom.py (hunk 1 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py:3233]: Yes, it is.

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py (hunk 2 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:773]: Service change. It returns "False" not "false"

Comment (yungezz) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:773]: can the test util be changed to support case-insensitive compare?

**Review by yungezz — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py (hunk 2 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:773]: The "False" I added means case sensitive = False. Default value is True.
